fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/PickupItemCallback.java

```diff
@@ -0,0 +1,35 @@
+package net.fabricmc.fabric.api.event.player;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.ItemEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.ActionResult;
+import net.minecraft.world.World;
+
+/**
+ * Callback for picking up an item.
+ * Is hooked in before the item is picked up.
+ * <p>
+ * Upon return:
+ * - SUCCESS cancels further processing and picks up the item.
+ * - PASS falls back to further processing.
+ * - FAIL cancels further processing and does not pick up the item.
```

Comment:
The documented behavior does not match the implementation. `SUCCESS` and `FAIL` currently both have the effect of `FAIL`. If this is intended, the ActionResult should be turned into a boolean. Otherwise, the mixin should be updated.

Reply:
^^^

Reply:
Fixed. The event now only cancels if FAIL is returned.